Aido_pridejimas_garsui.py

```diff
@@ -0,0 +1,76 @@
+import matplotlib.pyplot as plt
+import easygui
+import soundfile as sf
+import pandas as pd
+
+
+def openFile():  # leidžia pasirinki wav faila
+    file_path = easygui.fileopenbox()
+    return file_path
+
+
+def stereo(data):
+    # patikrina ar reiksmei galima rasti ilgi
+    return hasattr(data[0], "__len__")
+
+
+def getTmax(data):
+    if (stereo(data)):
+        data = [item for sublist in data for item in sublist]
+    return max(max(data), -min(data))
+
+
+def normalizeValues(data):
+    tmax = getTmax(data)
+
+    if (stereo(data)):
+        return [[elem/tmax for elem in sublist] for sublist in data]
+    return [elem/tmax for elem in data]
+
+
+def visualization(elements, indexes):
+    plt.figure()
+    if (stereo(elements)):
+        df = pd.DataFrame({'Left': [sublist[0] for sublist in elements], 'Right': [
+                          sublist[1] for sublist in elements]}, index=indexes)
+        df.plot()
+    else:
+        df = pd.Series(elements, index=indexes)
+        df.plot()
+
+    plt.ylabel('Reikšmės')
+    plt.xlabel('Laikas')
+
+
+def addEcho(sound_data, rate, echo_volume, delay):
+    echo_data = []
+    # indeksas, nuo kurio pridedamas aidas
+    index_delay = int((delay / 1000) * rate)
```

Comment:
delay atrodo kaip nereikalingas kintamasis, nes jis taip pat yra jau įrašytas į kodą, dėl to galima sutaupyti kodo ir vietoj delay/1000 rašyti 0.2